modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/edit_cp_definition_diagram_setting.jsp

```diff
@@ -28,6 +28,8 @@ String type = BeanParamUtil.getString(cpDefinitionDiagramSetting, renderRequest,
 CPDefinitionDiagramType cpDefinitionDiagramType = cpDefinitionDiagramSettingDisplayContext.getCPDefinitionDiagramType(type);
 %>
 
+<link href="<%= PortalUtil.getStaticResourceURL(request, PortalUtil.getPathModule() + "/commerce-shop-by-diagram-web/css/main.css") %>" rel="stylesheet" />
+
```

Comment:
Why is a `link` tag here? 

(1) Adding a `link` tag in `body` is a bad practice. It means page will re-render twice, causing performance and FOUC issues.
(2) In Liferay we have standard ways to include CSS files, and this is not one of them. A module-scoped CSS file is most commonly added through [com.liferay.portlet.header-portlet-css property](https://github.com/liferay/liferay-portal/blob/master/modules/apps/account/account-admin-web/src/main/java/com/liferay/account/admin/web/internal/portlet/AccountEntriesAdminPortlet.java#L38).